config/config.go

```diff
@@ -312,135 +243,117 @@ type IstioComponentNamespaces map[string]string
-	c.ExternalServices.Grafana.InClusterURL = strings.TrimSpace(getDefaultString(EnvGrafanaInClusterURL, ""))
-	c.ExternalServices.Grafana.URL = strings.TrimSpace(getDefaultString(EnvGrafanaURL, ""))
-	c.ExternalServices.Grafana.Auth = getAuthFromEnv("GRAFANA")
-
-	// Tracing Configuration
-	c.ExternalServices.Tracing.Enabled = getDefaultBool(EnvTracingEnabled, true)
-	c.ExternalServices.Tracing.Path = ""
-	c.ExternalServices.Tracing.InClusterURL = strings.TrimSpace(getDefaultString(EnvTracingInClusterURL, ""))
-	c.ExternalServices.Tracing.URL = strings.TrimSpace(getDefaultString(EnvTracingURL, ""))
-	c.ExternalServices.Tracing.Namespace = strings.TrimSpace(getDefaultString(EnvTracingServiceNamespace, getIstioComponentNamespace("tracing", c.IstioNamespace, c.IstioComponentNamespaces)))
-	c.ExternalServices.Tracing.Port = getDefaultInt32(EnvTracingServicePort, 16686)
-	c.ExternalServices.Tracing.Auth = getAuthFromEnv("TRACING")
-
-	c.ExternalServices.Istio.IstioIdentityDomain = strings.TrimSpace(getDefaultString(EnvIstioIdentityDomain, "svc.cluster.local"))
-	c.ExternalServices.Istio.IstioSidecarAnnotation = strings.TrimSpace(getDefaultString(EnvIstioSidecarAnnotation, "sidecar.istio.io/status"))
-	c.ExternalServices.Istio.UrlServiceVersion = strings.TrimSpace(getDefaultString(EnvIstioUrlServiceVersion, "http://istio-pilot:8080/version"))
-
-	// ThreeScale ConfigEnvKubernetesCacheNamespacesuration
-	c.ExternalServices.ThreeScale.AdapterName = strings.TrimSpace(getDefaultString(EnvThreeScaleAdapterName, "threescale"))
-	c.ExternalServices.ThreeScale.AdapterService = strings.TrimSpace(getDefaultString(EnvThreeScaleServiceName, "threescale-istio-adapter"))
-	c.ExternalServices.ThreeScale.AdapterPort = strings.TrimSpace(getDefaultString(EnvThreeScaleServicePort, "3333"))
-
-	// Token-based authentication Configuration
-	c.LoginToken.SigningKey = strings.TrimSpace(getDefaultString(EnvLoginTokenSigningKey, "kiali"))
-	c.LoginToken.ExpirationSeconds = getDefaultInt64(EnvLoginTokenExpirationSeconds, 24*3600)
-
-	// Kubernetes client Configuration
-	c.KubernetesConfig.Burst = getDefaultInt(EnvKubernetesBurst, 200)
-	c.KubernetesConfig.QPS = getDefaultFloat32(EnvKubernetesQPS, 175)
-	c.KubernetesConfig.CacheEnabled = getDefaultBool(EnvKubernetesCacheEnabled, false)
-	c.KubernetesConfig.CacheDuration = getDefaultInt(EnvKubernetesCacheDuration, 5*60)
-	c.KubernetesConfig.CacheNamespaces = getDefaultStringArray(EnvKubernetesCacheNamespaces, ".*")
-	c.KubernetesConfig.CacheIstioTypes = getDefaultStringArray(EnvKubernetesCacheIstioTypes, "VirtualService,DestinationRule,Gateway,ServiceEntry")
-	c.KubernetesConfig.CacheTokenNamespaceDuration = getDefaultInt(EnvKubernetesCacheTokeNamespaceDuration, 10)
-	c.KubernetesConfig.ExcludeWorkloads = getDefaultStringArray(EnvKubernetesExcludeWorkloads, "CronJob,DeploymentConfig,Job,ReplicationController,StatefulSet")
-
-	trimmedExclusionPatterns := []string{}
```

Comment:
This logic is being removed without an equivalent

Reply:
This is one reason why this PR is necessary - the env var default stuff was confusing and rarely used. If you notice, this code was actually virtually useless because it didn't do anything except *if* the exclusion list was specified by an environment variable *and* the same setting wasn't found in the ConfigMap yaml. Well, it never was the former (we never overrode via env var) and since the operator came online it never was the latter either (because this setting is always found in the Kiali CR so even if an env var was specified, it is never used - again, that's the problem with this env var override stuff - it isn't actually getting used now).

So, effectively, this was dead code and so no equivalent is needed.